### PR TITLE
PHP 8 support

### LINKED ISSUE #6: PHP 8.0 support
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes

#### Summary

To be prepared for the december release of PHP 8.0, this repository has some additional TODOs to be tested against the new major version.

In order to make this repository compatible, one has to follow these steps:

- [ ] Modify `composer.json` to provide support for PHP 8.0 by adding the constraint `~8.0.0`
- [ ] Modify `composer.json` to drop support for PHP less than 7.3
- [ ] Modify `composer.json` to implement phpunit 9.3 which supports PHP 7.3+
- [ ] Modify `.travis.yml` to ignore platform requirements when installing composer dependencies (simply add `--ignore-platform-reqs` to `COMPOSER_ARGS` env variable)
- [ ] Modify `.travis.yml` to add PHP 8.0 to the matrix (*NOTE*: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [ ] Modify source code in case there are incompatibilities with PHP 8.0

### PR DESCRIPTION
**Pull request by tanerkay**
Q 	A
New Feature 	yes

### Description

Addresses https://github.com/laminas/laminas-captcha/issues/6

Caveats:

- waiting on two upstream dependencies for PHP 8 support:
   - `laminas/laminas-math`
   - `laminas/laminas-recaptcha`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,42 +6,30 @@ cache:
 
 env:
   global:
-    - COMPOSER_ARGS="--no-interaction"
+    - COMPOSER_ARGS="--no-interaction --ignore-platform-reqs"
```
Comment (froschdesign) [inline .travis.yml]: Please revert this change and it to the related PHP version. You can compare with the Travis configuration of laminas-validator: https://github.com/laminas/laminas-validator/blob/23623b4756eab850829e5b5f6205eeaf1303e959/.travis.yml#L7-L38

**Comment by froschdesign** [comment]
@tanerkay 
Thanks for your contribution! 👍 

> waiting on two upstream dependencies for PHP 8 support…

In the meantime you can fix some other problems like:

```
PHP Fatal error:  Declaration of LaminasTest\Captcha\ImageTest::tearDown() must be compatible
with PHPUnit\Framework\TestCase::tearDown(): void
in /home/travis/build/laminas/laminas-captcha/test/ImageTest.php on line 65
```
https://travis-ci.com/github/laminas/laminas-captcha/jobs/470269060#L473

Thanks in advance!

File: composer.json [hunk 2]
```diff
@@ -21,18 +21,18 @@
...
+        "php": "^7.3 || ~8.0.0",
         "laminas/laminas-math": "^2.7 || ^3.0",
-        "laminas/laminas-stdlib": "^3.2.1",
-        "laminas/laminas-zendframework-bridge": "^1.0"
+        "laminas/laminas-stdlib": "^3.3",
+        "laminas/laminas-zendframework-bridge": "^1.1"
     },
     "require-dev": {
-        "laminas/laminas-coding-standard": "~1.0.0",
+        "laminas/laminas-coding-standard": "^2.1.1",
```
Comment (froschdesign) [inline composer.json]: I would suggest that the coding standard should be addressed in another pull request. This keeps the current pull request easier.

Comment (tanerkay) [reply composer.json]: Sure. I'll remember that for future similar PRs. I incorrectly assumed `1.0.0` had a PHP 7 constraint but turns out it doesn't have any so all good

**Comment by sivaschenko** [comment]
Hi @tanerkay @froschdesign are there any expectation when the PHP 8 support could be available?

Looks like `laminas/laminas-math` is already compatible with PHP 8
And https://github.com/laminas/laminas-recaptcha/pull/5 is ready as well
